securedrop/debian/app-code/usr/bin/securedrop-set-redis-auth.py

```diff
@@ -0,0 +1,125 @@
+from typing import Optional
+
+CONFIG_PY = Path("/var/www/securedrop/config.py")
+RQ_CONFIG_PY = Path("/var/www/securedrop/rq_config.py")
+REDIS_CONF = Path("/etc/redis/redis.conf")
+
+PYTHON_RE = re.compile(r"^REDIS_PASSWORD = ['\"](.*?)['\"]$")
+REDIS_CONF_RE = re.compile(r"^requirepass (.*?)$")
+
+
+def read_python_file(path: Path) -> Optional[str]:
+    """Extract the password from a Python file"""
+    if not path.exists():
+        # rq_config.py might not exist yet
+        return None
+    contents = path.read_text()
+    # Read in reverse because we want to look for the last matching line
+    # since it'll take precedence in Python
+    for line in contents.splitlines()[::-1]:
+        match = PYTHON_RE.match(line)
+        if match:
+            return match.group(1)
+    # Nothing found
+    return None
+
+
+def write_python_file(path: Path, password: str) -> None:
+    """Set the new password in a Python file, removing any existing passwords"""
+    if path.exists():
+        lines = path.read_text().splitlines()
+    else:
+        lines = []
+    # Take the existing file, remove any matching password lines, and then add
+    # our new password at the end
+    lines = [line for line in lines if not PYTHON_RE.match(line)]
+    lines.append(f"REDIS_PASSWORD = '{password}'")
+    if path == RQ_CONFIG_PY and not path.exists():
+        # Ensure rq_config.py is created with the correct permissions
+        path.write_text("")
+        path.chmod(0o640)
+        shutil.chown(path, "root", "www-data")
+
+    path.write_text("\n".join(lines) + "\n")
+
+
+def read_redis_conf(path: Path) -> Optional[str]:
+    """Extract the password from our redis.conf file"""
+    contents = path.read_text()
+    # Read in reverse because we want to look for the last matching line
+    # since redis uses the last requirepass stanza
+    for line in contents.splitlines()[::-1]:
+        match = REDIS_CONF_RE.match(line)
+        if match:
+            return match.group(1)
+    # Nothing found
+    return None
+
+
+def write_redis_conf(path: Path, password: str) -> None:
+    """Set the new password in a redis.conf file, removing any existing passwords"""
+    lines = path.read_text().splitlines()
+    # Take the existing file, remove any matching password lines, and then add
+    # our new password at the end
+    lines = [line for line in lines if not REDIS_CONF_RE.match(line)]
+    lines.append(f"requirepass {password}")
+    path.write_text("\n".join(lines) + "\n")
+
+
+def generate_password() -> str:
+    """
+    Generate a base64-encoded, 32-byte random string
+
+    This is roughly equivalent to `head -c 32 /dev/urandom | base64`
+    """
+    return secrets.token_urlsafe(32)
+
+
+def check() -> bool:
+    config_py = read_python_file(CONFIG_PY)
+    rq_config_py = read_python_file(RQ_CONFIG_PY)
+    redis_conf = read_redis_conf(REDIS_CONF)
+    return config_py is not None and config_py == rq_config_py and rq_config_py == redis_conf
```

Comment:
Nit: I found the 1+2 conditionals confusing at first; how about:

```suggestion
    configs = {config_py, rq_config_py, redis_conf}
    if None in configs:
        return False
    return len(configs) == 1
```